### ISSUE TITLE
Add new config setting muc_show_room_info 

### ISSUE THREAD

**Issue by deleolajide** [conversejs/converse.js#1998]
**Is your feature request related to a problem? Please describe.**

When CSS is used to suppress unwanted info messages, multiple lines displaying the date of the days occur even when no conversation occurred. The date should only be displayed if a conversation has occurred on that day, but because the info messages that triggered the data change are suppressed. This weird UI situation occurs.
![image](https://user-images.githubusercontent.com/110731/80242525-643cad00-865d-11ea-9521-680d26e011d1.png)

See https://git.startinblox.com/components/sib-chat/issues/121 for original issue report

**Describe the solution you'd like**
Create a new configuration setting called **muc_show_room_info ** to suppress the info messages  and do this
```
if (_converse.api.settings.get("muc_show_room_info")) {
    this.messages.create({
      'type': 'info',
      message
    });
}

```

**Comment by jcbrand** [comment]
In trunk, many info messages are now stored in the `notifications` data structure on the ChatRoom.

I think we can change the `muc_show_join_leave` setting to take a list of types that should be shown or not.

Existing types for these notifications are `op`, `deop`, `entered`, `exited`, `mute` and `voice`.

**Comment by deleolajide** [comment]
> I think we can change the `muc_show_join_leave` setting to take a list of types that should be shown or not

We may to change the name otherwise, it will become confusion many months from now. I appreciate you are reluctant to add yet another config setting, but this is a simple and straightforward case with minimum impact elsewhere. 

In the early days of pade, it was the one thing everyone complained about and I added a config setting to suppress info messages a while back now.

**Comment by deleolajide** [comment]
You are right. Having two settings to control showing info messages is also confusing. This looks like another code refactor for Converse 7.0. 

I will implement a quick fix for 6.x using **muc_show_room_info** and then come back later to refactor **muc_show_join_leave** from a boolean into list of types to be shown or not.
